djactasauth/backends.py

```diff
@@ -60,19 +60,33 @@ def authenticate(self, username=None, password=None, **kwargs):
                 username=username, password=password, **kwargs)
 
     def _authenticate(self, username=None, password=None, **kwargs):
-        if self.is_act_as_username(username):
-            auth_username, act_as_username = username.split(self.sepchar)
-        else:
-            auth_username = act_as_username = username
-
-        auth_user = super(ActAsModelBackend, self)._authenticate(
-                username=auth_username, password=password, **kwargs)
-        if not auth_user:
-            return auth_user
-        if auth_username != act_as_username:
-            UserModel = get_user_model()
+        self.fail_unless_one_aaa_backend_is_configured()
+        assert password is not None
```

Comment:
it was one of one of those hard to track down bugs. In Django 1.11, `request` was added as an extra first arg to `authenticate`, and I didn't catch that. This method is usually called through the ui, which means, even if it is empty, it should only supply an empty string for password, hence I thought leaving the assertion in won't cause any problems. 

I'm open to suggestions to add a better regression test for it though! :)